presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -89,6 +92,18 @@ public RuleAssert withSession(Session session)
         return this;
     }
 
+    public RuleAssert withStats(String nodeId, PlanNodeStatsEstimate nodeStats)
+    {
+        ((TestingStatsCalculator) statsCalculator).stats.put(new PlanNodeId(nodeId), nodeStats);
+        return this;
+    }
+
+    public RuleAssert withStatsCalculator(StatsCalculator statsCalculator)
+    {
+        this.statsCalculator = requireNonNull(statsCalculator, "statsCalculator is null");
```

Comment:
Wrap with `TestingStatsCalculator`?

Reply:
The idea it that only of `withStats(), withStatsCalculator()` should be used.
If i can `withStatsCalculator(..).withStats(..)` then also I should be able to `.withStats(..).withStatsCalculator(..)` (because: why not?), and then wrapping is not enough.

Reply:
this would be easier to weigh in on if there were examples with usages, but right now I don't see why we wouldn't want to be able to inject stats for any stats calculator.  (so agreeing with @losipiuk's suggestion to wrap with TestingStatsCalculator) .  And we could have withStatsCalculator() keep the stats map from the previous stats calculator, so order wouldn't matter anymore.

Reply:
ah, i see, indeed, the usage is still coming (it will be a stats calculator that dumbly returns UNKNOWN_STATS for every request). I'd prefer to leave it this way. If this is a problem now, I can remove this method from this PR. Up to you.

Reply:
yeah, I'd prefer to add this in the pr where it would be used.